File: vojaro.common/domain/Asignatura.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace vojaro.domain
{
	public class Asignatura : Entity
    {
        public string Codigo { get; set; }
        public string Nombre { get; set; }
        public int Anio { get; set; }
        public int Cuatrimestre { get; set; }
        public int CargaHoraria { get; set; }
        public int Creditos { get; set; }
        public bool EsInterdisciplinaria { get; set; }

        public virtual long CarreraId { get; set; }
        public virtual long? CarreraOrientacionId { get; set; }

        public virtual Carrera Carrera { get; set; }
        public virtual CarreraOrientacion CarreraOrientacion { get; set; }
        public virtual ICollection<Correlativa> Correlativas { get; set; }
    }
}

File: vojaro.web/ClientApp/src/app/api/models/asignatura.ts
import { Carrera } from "./carrera";
import { CarreraOrientacion } from "./carrera-orientacion";
import { Universidad } from "./universidad";

export interface Asignatura {
   id: number;
   codigo: string;
   nombre: string;
   anio: number;
   cuatrimestre: number;
   cargaHoraria: number;
   creditos: number;
   esInterdisciplinaria: boolean;

   correlativas: any[] | null;
   universidad: Universidad | null;
   carrera: Carrera | null;
   carreraOrientacion: CarreraOrientacion | null;

   correlativasId: number[] | null;
   universidadId: number;
   carreraId: number;
   carreraOrientacionId: number | null;
  }
